bittide/src/Bittide/ClockControl/Callisto.hs

```diff
@@ -125,17 +131,21 @@ callisto targetCount updateEveryNCycles dataCounts =
   k_i = pure 1e-11
   fStep = pure 5e-4
 
+  safePopCount :: Signal dom (BitVector n) -> Signal dom (Signed 32)
+  safePopCount v = case useLowerLimit @n @m @32 of
+    Dict -> sumTo32 <$> (map (unpack . pack) . bv2v <$> v)
```

Comment:
Let's move this function to `Bittide.ClockControl.Callisto.Util`, it might come in handy somewhere else. It's implementation can be a bit simpler too (don't lift to `Signal`, unpack straight into the `Vec` we need):

```haskell
popCountTo32 ::
  forall n .
  ( KnownNat n
  , (1 + n) <= 32
  , 1 <= n
  ) =>
  BitVector n ->
  Signed 32
popCountTo32 =
  sumTo32 . unpack @(Vec n (Unsigned 1))
```

And after thinking about it some more, I'm still very unsold on this approach. We could squash all these constraints by simply implementing it as:

```haskell
popCountTo32 :: KnownNat n => BitVector n -> Signed 32
popCountTo32 = resize . bitCoerce . popCount
```

This makes the function much more convenient to use, and more efficient to compute. It will fail for `BitVector n` where `n > 2147483648` but this size is utterly non-sensical for hardware designs. We can still document this fact of course.

Reply:
Seems like your `popCountTo32` is not synthesizable: https://github.com/bittide/bittide-hardware/actions/runs/3912332689/jobs/6686876838. I'll try to revert to my safeVersion to see whether that fixes the problem.

Reply:
Yep, `sumTo32 . unpack @(Vec n (Unsigned 1))` works. Are you satisfied with keeping my `safePopCountTo32` then or do you still want to go with the non-synthesizable `popCount`?

Reply:
As discussed, we stay with `safePopCountTo32` for the moment.